build.gradle

```diff
@@ -1,78 +1,265 @@
+apply plugin: 'idea'
+apply plugin: "org.sonarqube"
+apply plugin: 'org.springframework.boot'
+apply plugin: 'io.spring.dependency-management'
+apply plugin: 'com.github.ben-manes.versions'
+apply plugin: 'info.solidsoft.pitest'
 
 def buildNumber = System.getenv("RELEASE_VERSION")?.replace("refs/tags/", "") ?: "1.0.0"
 
 group 'com.github.hmcts'
 
 version buildNumber
 
-def gitRepo = "https://github.com/hmcts/befta-fw"
-
-def pomConfig = {
-    licenses {
-        license {
-            name "MIT License"
-            url "http://www.opensource.org/licenses/mit-license.php"
-            distribution "repo"
-        }
-    }
-
-    scm {
-        url "https://github.com/hmcts/civil-sdt.git"
-    }
-}
+def gitRepo = "https://github.com/hmcts/civil-sdt-common.git"
 
 java {
     toolchain {
         languageVersion = JavaLanguageVersion.of(17)
     }
 }
 
-repositories {
-    // Use JCenter for resolving dependencies.
-    jcenter()
-    mavenLocal()
-    mavenCentral()
-    maven {
-        url 'https://jitpack.io'
+ext {
+    junitVersion           = '5.7.2'
+    junitPlatformVersion   = '1.7.2'
+    log4jVersion = '2.20.0'
+    testcontainers = '1.17.5'
+    cxfVersion = '3.5.4'
+    jaxwsVersion = "2.3.1"
+}
+
+ext.libraries = [
+        junit5: [
+                "org.junit.jupiter:junit-jupiter-api:${junitVersion}",
+                "org.junit.jupiter:junit-jupiter-engine:${junitVersion}",
+                "org.junit.jupiter:junit-jupiter-params:${junitVersion}",
+                "org.junit.platform:junit-platform-commons:${junitPlatformVersion}",
+                "org.junit.platform:junit-platform-engine:${junitPlatformVersion}"
+        ]
+]
+
+task sourcesJar(type: Jar, dependsOn: classes) {
+    classifier = 'sources'
+    from sourceSets.main.allSource
+}
+
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier = 'javadoc'
+    from javadoc.destinationDir
+}
+
+task testJar(type: Jar, dependsOn: testClasses) {
+    classifier = 'test'
+    from sourceSets.test.output
+}
+
+tasks.withType(Test) {
+    useJUnitPlatform()
+
+    testLogging {
+        exceptionFormat = 'full'
     }
 }
 
-jar {
-    baseName = "civil-sdt-common"
+artifacts {
+    archives sourcesJar
+    archives javadocJar
+    archives testJar
 }
 
+bootJar {
+    enabled = false
+}
 
 dependencies {
-    // Use JUnit Jupiter API for testing.
-    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.2'
 
-    // Use JUnit Jupiter Engine for testing.
-    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine'
+    // This dependency is used internally, and not exposed to consumers on their own compile classpath.
+    implementation 'com.google.guava:guava:30.0-jre'
+    implementation group: 'org.apache.logging.log4j', name: 'log4j-api', version: log4jVersion
+    implementation group: 'org.apache.logging.log4j', name: 'log4j-core', version: log4jVersion
+    implementation 'org.apache.logging.log4j:log4j-slf4j2-impl:2.20.0'
```

Comment:
Should this use the log4jVersion property for the version number instead of the hard coded value?

Reply:
done

Reply:
Confirmed